tests/common/widgets/breadcrumb/options.ts

```diff
@@ -0,0 +1,298 @@
+import {
+  createSearchClient,
+  createMultiSearchResponse,
+  createSingleSearchResponse,
+} from '@instantsearch/mocks';
+import type { BreadcrumbSetup } from '.';
+import type { Act } from '../../common';
+import { screen } from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
+import { wait } from '@instantsearch/testutils';
+
+const hierarchicalFacets = {
+  'hierarchicalCategories.lvl0': {
+    'Cameras & Camcorders': 1369,
+  },
+  'hierarchicalCategories.lvl1': {
+    'Cameras & Camcorders > Digital Cameras': 170,
+  },
+};
+
+const hierarchicalAttributes = Object.keys(hierarchicalFacets);
+
+const searchClient = createSearchClient({
+  search: jest.fn((requests) => {
+    return Promise.resolve(
+      createMultiSearchResponse(
+        ...requests.map(() =>
+          createSingleSearchResponse({
+            facets: hierarchicalFacets,
+          })
+        )
+      )
+    );
+  }),
+});
+
+/**
+ * Removes rendered DOM discrepancies between flavors when
+ * they don't affect the behavior of the widget
+ * @param element A DOM element to normalize
+ * @returns A normalized DOM element
+ */
+function normalizeSnapshot(element: Element | null): Element | null {
```

Comment:
not sure if it's needed, but this is "more consistent" I think

```suggestion
function normalizeSnapshot<TElement extends Element | null>(
  element: TElement
): TElement {
```

That way the element doesn't become optional if it was already asserted

Reply:
In the end it caused issues with TypeScript. I opted for a more simplistic approach, as there's no reason for the `element` to be null when we want to normalize for snapshot. See https://github.com/algolia/instantsearch/pull/5674/commits/635991e942c11f87099e50d1fe28ede68094fa83.